internal/core/src/mmap/ChunkData.h

```diff
@@ -0,0 +1,199 @@
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+#include "common/Array.h"
+#include "storage/MmapChunkManager.h"
+namespace milvus {
+/**
+ * @brief FixedLengthChunk
+ */
+template <typename Type>
+struct FixedLengthChunk {
+ public:
+    FixedLengthChunk() = delete;
+    explicit FixedLengthChunk(const uint64_t size,
+                              storage::MmapChunkDescriptor descriptor)
+        : mmap_descriptor_(descriptor), size_(size) {
+        auto& mmap_manager = mmap_descriptor_->second;
+        milvus::ErrorCode err_code;
+        data_ = (Type*)(mmap_manager->Allocate(
+            mmap_descriptor_->first, sizeof(Type) * size, err_code));
+        AssertInfo(data_ != nullptr,
```

Comment:
Unify the error reports. I noticed 3 places are using error code and this one working with nullptr check